google/drive/apis/files.py

```diff
@@ -320,6 +322,19 @@ def download_file(service: Resource, file_id: str) -> Optional[bytes]:
             }
             export_mime_type = export_formats.get(file["mimeType"], "application/pdf")
 
+            # Add appropriate file extension based on export format
+            extension_map = {
+                "application/pdf": ".pdf",
+                "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet": ".xlsx",
+                "application/vnd.openxmlformats-officedocument.wordprocessingml.document": ".docx",
+                "application/vnd.openxmlformats-officedocument.presentationml.presentation": ".pptx",
+            }
+
+            # Add extension if the file doesn't already have it
+            extension = extension_map.get(export_mime_type, ".pdf")
+            if not file_name.lower().endswith(extension):
+                file_name = f"{file_name}{extension}"
```

Comment:
Is there some kind of MIME type-to-file extension library we could use instead? It seems weird to limit it to only these four types (and to default to PDF).

Reply:
it's a subset of https://developers.google.com/workspace/drive/api/guides/ref-export-formats. I can't find a library for it.